default.nix

```diff
@@ -52,7 +52,12 @@ let
     override = "stackage";
   });
 
-  packages = self: ({
```

Comment:
I keep wondering if we can't simplify this. Do we really need to bring in the callPackage logic for the buildPackages?

Reply:
The callPackages stuff needs to be *built* with buildPackages, otherwise if you do a cross-compile you wind up building nix-tools for the host platform rather than the build platform. This actually escaped my notice at first because I had binfmt emulation for armv7l set up on my build machine, but popped up when I tried an Aarch64 cross compile. As for why it's in the `packages` fixpoint...that's where everything else was so I stuck it there.

Reply:
Can't we just pull nix-tools from pkgs.buildPackages.callPackage ./nix-tools { inherit fetchExternal; };